src/main/java/com/fauna/codec/UTF8FaunaParser.java

```diff
@@ -261,15 +262,15 @@ private void handleTaggedString(FaunaTokenType token) throws IOException {
 
     private void advanceTrue() {
         if (!advance()) {
-            throw new ClientException("Unexpected end of underlying JSON reader.");
+            throw new PartialDecodeException("Unexpected end of underlying JSON reader.");
```

Comment:
because we have a separate task to deal with ClientException everywhere, let's exclude them from this PR where they already exist

Reply:
I backed `PartialDecodeException` out of this PR.